pkg/knuu/instance.go

```diff
@@ -18,31 +18,44 @@ import (
 
 // Instance represents a instance
 type Instance struct {
-	name                  string
-	imageName             string
-	k8sName               string
-	state                 InstanceState
-	instanceType          InstanceType
-	kubernetesService     *v1.Service
-	builderFactory        *container.BuilderFactory
-	kubernetesStatefulSet *appv1.StatefulSet
-	portsTCP              []int
-	portsUDP              []int
-	command               []string
-	args                  []string
-	env                   map[string]string
-	volumes               []*k8s.Volume
-	memoryRequest         string
-	memoryLimit           string
-	cpuRequest            string
-	policyRules           []rbacv1.PolicyRule
-	livenessProbe         *v1.Probe
-	readinessProbe        *v1.Probe
-	startupProbe          *v1.Probe
-	files                 []*k8s.File
-	isSidecar             bool
-	parentInstance        *Instance
-	sidecars              []*Instance
+	name                     string
+	imageName                string
+	k8sName                  string
+	state                    InstanceState
+	instanceType             InstanceType
+	kubernetesService        *v1.Service
+	builderFactory           *container.BuilderFactory
+	kubernetesStatefulSet    *appv1.StatefulSet
+	portsTCP                 []int
+	portsUDP                 []int
+	command                  []string
+	args                     []string
+	env                      map[string]string
+	volumes                  []*k8s.Volume
+	memoryRequest            string
+	memoryLimit              string
+	cpuRequest               string
+	policyRules              []rbacv1.PolicyRule
+	livenessProbe            *v1.Probe
+	readinessProbe           *v1.Probe
+	startupProbe             *v1.Probe
+	files                    []*k8s.File
+	isSidecar                bool
+	parentInstance           *Instance
+	sidecars                 []*Instance
+	ingress                  *Ingress
```

Comment:
maybe is adding more complexity, did you consider splitting the obsy values in another struct?

Reply:
Agreed with Jose here. Please consider avoiding `hero` structs or funcs. 
You can avoid either by duck-typing or you can get inspired by node repo with using Options / Builders pattern

Reply:
https://github.com/celestiaorg/knuu/pull/174/commits/0dd1a454e3b05f7cfdf2b6c5a62df15aa53194fd